spec/socialprovider.spec.js

```diff
@@ -151,4 +153,55 @@ describe("Tests for message batching in Social provider", function() {
     expect(xmppSocialProvider.vCardStore.updateProperty)
         .toHaveBeenCalledWith('id', 'status', 'OFFLINE');
   });
+
+  it('disconnects when no reply to ping', function() {
+    spyOn(window.XMPP, 'Client').and.returnValue(xmppClient);
+    spyOn(xmppSocialProvider, 'logout');
+    xmppSocialProvider.connect();
+    xmppSocialProvider.ping_();
+    jasmine.clock().tick(xmppSocialProvider.MAX_MS_PING_REPSONSE_ + 10);
+    expect(xmppSocialProvider.logout).toHaveBeenCalled();
+  });
+
```

Comment:
I'm not sure we can write a test that causes a setInterval callback set for
1000 ms to be called with 2000ms passed since the last call. The only way
I've been able to test that code is to let my computer go to sleep and then
start it again

On Fri, Feb 6, 2015, 19:24 salomegeo <notifications@github.com> wrote:

> In spec/socialprovider.spec.js
> <https://github.com/freedomjs/freedom-social-xmpp/pull/93#discussion_r24282799>
> :
>
> > @@ -151,4 +153,55 @@ describe("Tests for message batching in Social provider", function() {
> >      expect(xmppSocialProvider.vCardStore.updateProperty)
> >          .toHaveBeenCalledWith('id', 'status', 'OFFLINE');
> >    });
> > +
> > +  it('disconnects when no reply to ping', function() {
> > +    spyOn(window.XMPP, 'Client').and.returnValue(xmppClient);
> > +    spyOn(xmppSocialProvider, 'logout');
> > +    xmppSocialProvider.connect();
> > +    xmppSocialProvider.ping_();
> > +    jasmine.clock().tick(xmppSocialProvider.MAX_MS_PING_REPSONSE_ + 10);
> > +    expect(xmppSocialProvider.logout).toHaveBeenCalled();
> > +  });
> > +
>
> Can you also test for 2s intervals?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom-social-xmpp/pull/93/files#r24282799>
> .
>